File: Coding Exercises/Python/Day54/is_harshad_number.py
def is_harshad_number(n: int) -> bool:
    if n <= 0:
        return False

    sum = 0
    originalN = n
    while n > 0:
        digit = n % 10
        sum += digit
        n = n // 10

    return originalN % sum == 0

File: Coding Exercises/TypeScript/Day54/isHarshadNumber.ts
function isHarshadNumber(n: number): boolean {
  if (n <= 0) {
      return false;
  }

  let sum = 0;
  let originalN = n;
  while (n > 0) {
      const digit = n % 10;
      sum += digit;
      n = Math.floor(n / 10);
  }

  return originalN % sum === 0;
}
